apps/labs/posts/packaging_painpoints_in_2021.md

```diff
@@ -0,0 +1,228 @@
+
+- Shipping JupyterLab 3.0 extensions without Node.js is kinda cool.
+- Rust/Maturin to develop Python packages is nice.
+    - The relocation of depending binaries must be done manually in some cases.
+
+
+## Topic 2: The conda world
+
+- Conda itself is almost unmaintained
+
+_RG: a quick peek at [the conda repo](https://github.com/conda/conda/pulse) indeed shows declining activity. Hopefully the new Conda Community organization will help generate new energy for `conda` contributions._
+
+- Mamba is still a little tricky/slow to install
+
+_RG: this is definitely true if you start with conda; it can take minutes for conda to do the solve to install mamba and its dependencies from conda-forge. With the recently added [Mambaforge](https://github.com/conda-forge/miniforge#mambaforge) installer that problem should be mostly solved now._
+
+- No TensorFlow 
+    - Does conda-forge count?
+
+_RG: this is actually no longer true, the TensorFlow package in `defaults` is reasonably but not completely up to date (2.3.0 CPU and GPU packages at the moment, missing 2.3.1-2.4.1). In general it's a problem indeed that important packages like TensorFlow and PyTorch in `defaults` may be one or even more versions behind._
+
+- Conda-forge is growing too large
+    - In which respect?
+
+_RG: As discussed in [this blog post on conda performance](https://www.anaconda.com/blog/how-we-made-conda-faster-4-7), the more packages there are in a channel, the slower the conda solves get. Given the ever-growing size of conda-forge, conda is getting slower and slower when used with conda-forge._
+
+- Why oh why are defaults and conda-forge still not using a 100% compatible toolchain??
+
+_RG: This would be so helpful. Not being able to mix `defaults` and `conda-forge`, and users doing it anyway because they need packages from both, is a long-standing issue._
+
+- Conda solves in CI often get really bogged down
+    - Also conda is never the default in CI so you need to do quite a bit of manipulation on some platforms
+- Confusion in the community over the new Anaconda ToS
+
+_RG: Anaconda has done the right thing here imho: make large institutional users pay if they use the `defaults` channel. Packaging is hard and labour-intensive, and having a sustainable revenue stream to offset the cost of packaging difficult libraries, maintaining core tooling like `conda-build`, and serving packages to millions of users is important. Conda-forge and all other channels are still completely free. Unfortunately Anaconda communicated that very poorly, which led to unnecessary confusion. See [this blog posts from the conda-forge team](https://conda-forge.org/blog/posts/January 24, 2021
```

Comment:
1. adding a closed bracket to fix md linking.
2. conda-forge has changed their website, so this link needs updating. @rgommers can you please have a look here and identify the blog you were referring to? 
https://conda-forge.org/blog/

```suggestion
_RG: Anaconda has done the right thing here imho: make large institutional users pay if they use the `defaults` channel. Packaging is hard and labour-intensive, and having a sustainable revenue stream to offset the cost of packaging difficult libraries, maintaining core tooling like `conda-build`, and serving packages to millions of users is important. Conda-forge and all other channels are still completely free. Unfortunately Anaconda communicated that very poorly, which led to unnecessary confusion. See [this blog posts from the conda-forge team](https://conda-forge.org/blog/posts/January 24, 2021)
```

Reply:
https://conda-forge.org/blog/posts/2020-11-20-anaconda-tos/